src/py21cmfast/src/PerturbField.c

```diff
@@ -257,13 +260,50 @@ int ComputePerturbField(
                         if (zi >= (dimension)){ zi -= (dimension);}
                         if (zi < 0) {zi += (dimension);}
 
+                        // Determine the fraction of the perturbed cell which overlaps with the 8 nearest grid cells,
+                        // based on the grid cell which contains the centre of the perturbed cell
+                        d_x = xf - (float)(xi);
+                        d_y = yf - (float)(yi);
+                        d_z = zf - (float)(zi);
+                        t_x = 1. - d_x;
+                        t_y = 1. - d_y;
+                        t_z = 1. - d_z;
+
+                        // Determine the grid coordinates of the 8 neighbouring cells
+                        xp1 = xi + 1;
+                        if(xp1 >= dimension) { xp1 -= (dimension);}
+                        yp1 = yi + 1;
+                        if(yp1 >= dimension) { yp1 -= (dimension);}
+                        zp1 = zi + 1;
+                        if(zp1 >= dimension) { zp1 -= (dimension);}
+
```

Comment:
I think these should be defined wrt the midpoint of the cell, e.g. d_x = fabs(xf - (float)(xi+0.5)), and  xp1  being either xi+1 or xi-1, depending on if xf is gtr or lessthen xi+0.5.

Reply:
I have implemented these changes